src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/PersonBlockingRulesRepository.kt

```diff
@@ -11,10 +14,16 @@ class PersonBlockingRulesRepository {
   @PersistenceContext
   private val entityManager: EntityManager? = null
 
-  fun findMatchCandidates(person: Person, personQuery: String): List<PersonEntity> {
+  fun findMatchCandidates(person: Person, personQuery: String, pageable: Pageable): Page<PersonEntity> {
     val query = entityManager!!.createNativeQuery(personQuery, PersonEntity::class.java)
     query.setParameter("firstName", person.firstName)
     query.setParameter("lastName", person.lastName)
-    return query.resultList as List<PersonEntity>
+
+    // apply pagination
+    query.setFirstResult((pageable.offset.toInt()))
+
+    val results = query.resultList as List<PersonEntity>
+
+    return PageImpl(results, pageable, results.size.toLong())
```

Comment:
The `results.size.toLong()` needs to be the total amount of results we would get without paging. Where a `COUNT (*)` statement would come into play here